### ISSUE TITLE
Error: Timeout (2000) waiting for 127.0.0.1:9229 to be free

### ISSUE THREAD

**Issue by cristicmf** [nodejs/node-inspect#48]
Error: Timeout (2000) waiting for 127.0.0.1:9229 to be free
I have the same question

**Comment by jkrems** [comment]
Hey - thanks for the report! This should generally only happen if either a) you're using an older version of `node-inspect` or something else is genuinely blocking the port. The latter would happen if you'd try to start 2 CLI debuggers on the same machine.

Can you share an example script & command that was used to generate the error?

**Comment by Peripona** [comment]
there could also be the possibility, you are running `node debug` and try to run `node inspect` on another terminal.

**Comment by Michelle0609** [comment]
Hi ,I got the same issue. Does any one know how to fix that?
The script is very simple: 
var person={
    name:'Michelle'
};
person.age=25;
person.name='Mike';
console.log(person);

Command: node inspect app.js

**Comment by jkrems** [comment]
@Michelle0609 The most likely issue is that something else is running on port 9229. You can try running it on a different port using the following command:

```
node --inspect-port=9228 debug app.js
```

Alternatively you can try finding the thing blocking port 9229 on your machine. If neither of those options gets you anywhere, can you share the OS and version of node you're using?

**Comment by Michelle0609** [comment]
Hi Jan,
Thanks for the email. I tried "node --inspect-port=9228 debug app.js"

But got the following issue

(node:12276) [DEP0068] DeprecationWarning: `node debug` is deprecated.
Please use `node inspect` instead.
There was an internal error in node-inspect. Please report this bug.
Timeout (2000) waiting for 127.0.0.1:9228 to be free
Error: Timeout (2000) waiting for 127.0.0.1:9228 to be free
    at Timeout.setTimeout [as _onTimeout]
(node-inspect/lib/_inspect.js:63:14)
    at ontimeout (timers.js:478:11)
    at tryOnTimeout (timers.js:302:5)
    at Timer.listOnTimeout (timers.js:262:5)

My OS is windows 10 and node version is 9.1.0

Thanks
Michellle

**Comment by jkrems** [comment]
Thanks for the details!

Wonder if there are any important error cases [we're ignoring while trying](https://github.com/nodejs/node-inspect/blob/49d7f88e3466d5b4b46886cb1f997b2dfc123d8b/lib/_inspect.js#L74). I don't have Windows around right now but I might be able to try debugging it when I'm back in the office tomorrow.

**Comment by Michelle0609** [comment]
Thanks Jan!

**Comment by jorijnsmit** [comment]
I encountered this error and found out while trying to close my Terminal window that there was another instance of node running in it still. Probably did not close it properly and that's why the port was blocked.

**Comment by seancasaus** [comment]
To anyone making it to this thread with this issue, make sure you are exiting out of inspect using ctrl^c, not ctrl^z. If you use ctrl^z the debugger is still blocking the port and the only way to fix it is to exit and reenter terminal.

**Comment by blinking-in-the-light** [comment]
@Michelle0609 I got a good laugh from your example because it was identical enough to mine to know we were doing the same tutorial! Cheers.

**Comment by 14willh** [comment]
I'm not sure if anyone is still interested, but the solution is to use `node --inspect` instead of `node inspect`.

**Comment by 4rch1m3d35** [comment]
i'm having this issue as well and only have one terminal up still has same error 
Node.js:12.2.0
OS: Windows 10
on VSC

```
$ node inspect app.js
Timeout (2000) waiting for 127.0.0.1:9229 to be free
```

**Comment by kerryja** [comment]
Hey @4rch1m3d35 did you figure this out? I am having the same problem!

**Comment by 4rch1m3d35** [comment]
@kerryja nope nothing yet

**Comment by AleArmengol** [comment]
I'm having the same problem, did anyone figure this out? I already tried changing the port, but the error message is the same.
Timeout (2000) waiting for 127.0.0.1:9229 to be free
No matter what port I'm typing

**Comment by jkrems** [comment]
> I'm having the same problem, did anyone figure this out? I already tried changing the port, but the error message is the same.

@AleArmengol To clarify your comment: Do you mean the message literally keeps talking about `9229` or just that you keep seeing timeouts for the various ports? If the port in the error stays the same, can you share the command you are using to set different ports?

**Comment by 4rch1m3d35** [comment]
this is the command to change ports
`node --inspect=127.0.0.1:9230`

**Comment by jkrems** [comment]
To clarify, the command to start the CLI debugger on a custom port is `node inspect --port=9230 path/to/script.js`.

When using `node --inspect`, the error above shouldn't appear since that's the program being debugged, it's not starting the CLI debugger (`node-inspect`).

**Comment by victorgutemberg** [comment]
same issue here.

Node v10.13.0
Windows 10

Tried running in VSC and then another attempt in PowerShell.
Tried changing the ports and for all of then, the same error message appears.

`Timeout (2000) waiting for 127.0.0.1:12903 to be free`
`Timeout (2000) waiting for 127.0.0.1:25612 to be free`
`Timeout (2000) waiting for 127.0.0.1:5122 to be free`

**Comment by jkrems** [comment]
@victorgutemberg When you say "tried running in VSC", do you mean its integrated terminal? Also, can you share the exact command you ran?

**Comment by jkrems** [comment]
I wonder if this is something specific to Windows since so far that seems to be a common thread. I don't have a Windows machine anymore to test that on, so if somebody could do some debugging, it would be much appreciated. :)

**Comment by AleArmengol** [comment]
I ran this commands
node inspect --port=9229 app.js
Timeout (2000) waiting for 127.0.0.1:9229 to be free

node inspect --port=9230 app.js
Timeout (2000) waiting for 127.0.0.1:9230 to be free

**Comment by victorgutemberg** [comment]
@jkrems yeah, I meant the integrated terminal. I've tried using the default cdm and PowerShell, same result on both.

And the command that I was running is the same as just mentioned by @AleArmengol.

**Comment by pjgorman** [comment]
same problem Windows 10 Pro, Build 18362:
node inspect myScript.js gets "Timeout (2000) waiting for 127.0.0.1:9229 to be free".
Reproduced on various ports (9228, 9227)
Also, not specific to the VSC terminal.  Reproduced in windows cmd and in powershell

Seems to work fine on the Windows 7 machine

**Comment by hessproject** [comment]
I am also having this issue on different ports in Windows 10, both in VSC integrated terminal and regular powershell

**Comment by boba1987** [comment]
Had the same issue, the problem was that I was starting multiple instances of docker containers running node app, so the first instance that starts occupies `127.0.0.1:9229` and other instances cannot get to the port, thus trowing the error. After I set docker to start a single node app instance, error was gone

**Comment by 4rch1m3d35** [comment]
unfortunately i am not using docker and i am only starting one instance of the integrated terminal

**Comment by linqOxelar** [comment]
Same problem : Timeout (2000) waiting for 127.0.0.1:9229 to be free

Windows x64 18362.207
Node 12.6.0

**Comment by yilmazdurmaz** [comment]
node debug ...
node inspect ...
node-inspect ...

all 3 of them gives the same error. 
there is nothing using that port, nor firewall has any rules against. true for any port i tried with these command.

i tried with node v10.16.0 , v11.12.0 and v12.6.0 on latest Windows 10 1903. I have nothing installed to prevent ports, as port numbers are irrelevant as explained below.

i looked at the source code of node-inspect and saw "port=0" will open a random port. i tried command with "--port=0" and all above commands start at a random port without a problem. setting the port in chrome manually, debugging is also has no problem on chrome.

the problem is, exiting the debug process after a random port session, and trying to start a new session with appearently made available from previous session, this very same timeout error occurs.

in source code for "node-inspect"s "_inspect.js", i changed timeout to 5000 in "portIsFree" function, and voila, happy (!) ending. i am really not sure if this timeout is the real culprit behind the scenes, but nodejs developers should give it a try.

i am not much good on github forks and pull&push things, so can someone please do this change?

**Comment by kaluogh** [comment]
use --inspect-brk

**Comment by Aayuarch** [comment]
guys this works. just type node --inspect-brk "your filename.js" for eg: mine is app.js so
node --inspect-brk app.js

**Comment by yilmazdurmaz** [comment]
@kaluogh  @Aayuarch 
> use --inspect-brk

it is nice to know it. this is working.

but it is not the solution for the timeout problem. also it is not opening the debugger session in the same console, and also it is a blocking process on that console.

it is still possible to connect from another terminal or chrome, but that means extra work to do
(ie. >node inspect localhost:9229 )

there is an issue here about timeout on opening ports, and this must be fixed by at least giving the host pc more time to initialize ports. that is why i asked if someone can supply a fix with an increased timeout value.

**Comment by 10051999** [comment]
> @kaluogh  @Aayuarch
> 
> use --inspect-brk
> 
> it is nice to know it. this is working.
> BUT it is not the solution for the timeout problem. also it is not opening the debugger session in the same console, and more it is a blocking process on that console.
> it is still possible to connect from another terminal or chrome, but that means extra work to do
> (ie. >node inspect localhost:9229 )
> there is an issue here about timeout on opening ports, and this must be fixed by at least giving the host pc more time to initialize ports. that is why i asked if someone can supply a fix with an increased timeout value.

yeah bro the debugger sesson isn't opening

**Comment by 10051999** [comment]
> @kaluogh  @Aayuarch
> 
> use --inspect-brk
> 
> it is nice to know it. this is working.
> BUT it is not the solution for the timeout problem. also it is not opening the debugger session in the same console, and more it is a blocking process on that console.
> it is still possible to connect from another terminal or chrome, but that means extra work to do
> (ie. >node inspect localhost:9229 )
> there is an issue here about timeout on opening ports, and this must be fixed by at least giving the host pc more time to initialize ports. that is why i asked if someone can supply a fix with an increased timeout value.

This command is for the debugging of node file via chrome ……… type : node --inspect-brk app.js 
then open google chrome and type : chrome://inspect ………...Now you are all set to debug

**Comment by mikemaccana** [comment]
# In case it's unclear to anyone: this is not a matter of the port being in use. Using any port at all will result in the timeout error on Windows.

I know sometimes people on GitHub don't get what ports are and that they're in use. That isn't the case here.

`node inspect --port=9230 file.js` will fail.
`node inspect --port=9999 file.js` will fail.
`node inspect --port=3000 file.js` will fail.

The port is not in use. 

As others also note

**Comment by yilmazdurmaz** [comment]
I created a pull request, and until it is accepted or at least reviewed, please do the following manual patch:
- go to install folder of node's global packages. (it should be shown by ``npm config get prefix``)
- find ``node_modules\node-inspect\lib\`` and open ``_inspect.js`` file
- get to line 52, change ``timeout`` value to 9999 in ``portIsFree`` function

(also applies to issues #54 and #70)

**Comment by mikemaccana** [comment]
Hrm @yilmazdurmaz  I don't have `node-inspect` as a global npm module. Are you sure that's the right file?

**Comment by mikemaccana** [comment]
Thanks @yilmazdurmaz .

https://nodejs.org/en/docs/guides/debugging-getting-started/ says:

> node-inspect
> CLI Debugger supported by the Node.js Foundation which uses the Inspector Protocol.
> A version is bundled with Node and can be used with node inspect myscript.js.

But I can't find node-inspect - `npm config get prefix` shows me `C:\Users\mikem\AppData\Roaming\npm` but the only files in `node-modules` are npm itself (which makes sense as avoid global modules, installing buil tools locally and using `npx`). 

Do you know where the `node-inspect` bundled with node is installed?

**Comment by yilmazdurmaz** [comment]
> (which makes sense as I avoid global modules, installing build tools locally and using `npx` in most cases).
> 
> **Do you know where the `node-inspect` bundled with node is installed?**

I havent used ``"npx"`` before, but is seems you need to do ``"npm install package_name -D"`` prior to use it.

If so, that means it is a local installation, and ``npm-inspect`` should reside in ``"your_project_folder/node_modules/npm-inspect"`` folder. 

In other words, in step 1, go to any of your local projects' folder where you install&use node-inspect and continue steps in there.

Good Luck

**Comment by mikemaccana** [comment]
>  but is seems you need to do "npm install package_name -D" prior to use it.

Well yes - and thanks for trying to help - but that's nor really the point. I don't want to install an additional copy of `node-inspect`, I want to use and fix the version included in node, as mentioned at  https://nodejs.org/en/docs/guides/debugging-getting-started/ 

> A version is bundled with Node and can be used with node inspect myscript.js.

Ie, I want to run `node inspect` not `node-inspect`

**Comment by jkrems** [comment]
Unfortunately the code of the bundled version is part of the node binary (`node.exe`). It's not stored separately on disk. You could also try the following:

1. Run `node --inspect-brk your-script.js`.
1. Run `node inspect localhost:9229` in a separate terminal tab/window.

Restarting the target is now a bit more more annoying (it requires both restarting the actual script being debugged and then using the `restart` (or `r`) command in the `node inspect` repl to reconnect. But it may be easier as a workaround.

**Comment by yilmazdurmaz** [comment]
> Ie, I want to run `node inspect` not `node-inspect`

The thing is you (any other coder to have the same issue) have some options here:
- you wait for a fix until @jkrems is available to publish a fix
- you may try his suggestion above which is exhaustive
- or you can try (as I myself prefer) the third option mentioned in the same article you linked with the fix I described. 
  > The latest version can also be installed independently (e.g. ``"npm install -g node-inspect"``) and used with ``"node-inspect myscript.js"``

The choice is yours,
Cheers :)

**Comment by mikemaccana** [comment]
@yilmazdurmaz yeah I know the `--inspect-brk` workaround (and yes that works fine, thanks for suggesting it) but I was hoping to be able to land a patch in `node` itself (I still don't get if `node-inspect` the package is actually the same thing as `node inspect` since I assume the later may involve some native code.

**Comment by jkrems** [comment]
> (I still don't get if node-inspect the package is actually the same thing as node inspect since I assume the later may involve some native code).

The way it works is that the code for this library gets copied into the node source ([example PR](https://github.com/nodejs/node/pull/28039)) on a somewhat regular basis. Node then runs the code using its internal require system that also powers things like the JS bits of built-in modules like `fs`. So if a fix in `node-inspect` is merged, it can be forwarded to node itself fairly easily without any additional code.

**Comment by yilmazdurmaz** [comment]
@mikemaccana  Today, I am in a bit of debugger/patcher mood, so here is 4th, but surely unsupported method. After this, ``node inspect`` works without timeout error.
- find a hex editor, open ``node.exe``, search for ``portIsFree``, currently 2 results comes up, one of them has ``timeout = 2000`` nearby, replace it with a higher value, I suggest ``9999``.

While doing that, you will notice that Node.js actually holds packages almost as is. If you look at the beginning of the file, you will see the "native" codes. Native codes are only read by machine logic (assembler). In contrast, packages are hold as their human readable form. 

Actually I didn't expect this result. Anyone may apply this method on their free will, just should not expect support if something else gets broken.

**Comment by mikemaccana** [comment]
>  still don't get if node-inspect the package is actually the same thing as node inspect since I assume the later may involve some native code).

> find a hex editor, open node.exe, search for portIsFree, currently 2 results comes up, one of them has timeout = 2000 nearby, replace it with a higher value

Woo @yilmazdurmaz that was what I was confused about! So `node.exe` includes the JS itself. Mystery solved. Thanks! 💯

**Comment by harshalsojitra** [comment]
> 
> 
> use --inspect-brk

Thanks, it worked for me!

**Comment by gauthamgauda** [comment]
> use --inspect-brk

not working for me... :(

**Comment by mikemaccana** [comment]
@gauthamgauda ... and? what happens when you try?

**Comment by vikingdv** [comment]
@yilmazdurmaz 
Thanks! I opened node.exe in a hex editor and changed delay in portIsFree from 2000 to 3000. That was enough. Your comment was very helpful.
Windows 10 x64
Node 10.16

**Comment by 95rajasekar** [comment]
Anyone knows whats the difference between **node inspect app.js** & **node --inspect app.js**
I was trying the first one and was facing the same Timeout issue. When i tried the latter it worked fine.

**Comment by jkrems** [comment]
> Anyone knows whats the difference between `node inspect app.js` & `node --inspect app.js`

Yeah, those two commands look more similar than they should. The difference is:

* `node --inspect`: Run `app.js` but expose the remote debugging interface that things like vscode, Chrome devtools, Intellij, etc. can attach to. Otherwise `app.js` will run just like always and have full control over the terminal.
* `node inspect`: Start the interactive CLI debugger that's bundled with node. It's an alternative for visual debuggers like vscode or Chrome devtools. The CLI debugger will then use `node --inspect` to start the script in a separate process and attaches to it.

The relationship between the two is more obvious in the more explicit flow that also supports remote debugging across machines:

```bash
# Start app.js with the inspect interface exposed, by default on port 9229:
node --inspect app.js
# In a separate terminal, start the inspect client against port 9229:
node inspect -p 9229
```

Now the CLI debugger (`node inspect`) is connected to the node process with the debug interface enabled (`node --inspect`). Hope that helps. :)

**Comment by 95rajasekar** [comment]
> > Anyone knows whats the difference between `node inspect app.js` & `node --inspect app.js`
> 
> Yeah, those two commands look more similar than they should. The difference is:
> 
> * `node --inspect`: Run `app.js` but expose the remote debugging interface that things like vscode, Chrome devtools, Intellij, etc. can attach to. Otherwise `app.js` will run just like always and have full control over the terminal.
> * `node inspect`: Start the interactive CLI debugger that's bundled with node. It's an alternative for visual debuggers like vscode or Chrome devtools. The CLI debugger will then use `node --inspect` to start the script in a separate process and attaches to it.
> 
> The relationship between the two is more obvious in the more explicit flow that also supports remote debugging across machines:
> 
> ```shell
> # Start app.js with the inspect interface exposed, by default on port 9229:
> node --inspect app.js
> # In a separate terminal, start the inspect client against port 9229:
> node inspect -p 9229
> ```
> 
> Now the CLI debugger (`node inspect`) is connected to the node process with the debug interface enabled (`node --inspect`). Hope that helps. :)

Thanks! got the diff b/w the two 👍

**Comment by Haisum92** [comment]
> guys this works. just type node --inspect-brk "your filename.js" for eg: mine is app.js so
> node --inspect-brk app.js @ @ @ @yilmazdurmaz @kerryja

```Starting inspector on 127.0.0.1:9229 failed: address already in use
node: ../deps/uv/src/unix/core.c:117: uv_close: Assertion `!(((handle)->flags & (UV_CLOSING | UV_CLOSED)) != 0)' failed.
Aborted (core dumped)```

**Comment by yilmazdurmaz** [comment]
@jkrems  is there a way you may put a summary at the end and then put this discussion on a hold? it is too long that people miss the important parts we talked about.

**Comment by Prottoy2938** [comment]
I have tried `--inspect-brk app.js`,  it works,
But I can't debug in crome when using this method. I press the play button in crome inspect, and nothing happens in the even I had typed `debugger` in my code.

And after typing `--inspect-brk app.js` the timeout error is happening again. So have anyone solved this issue?  How do I get out of this timeout error?

**Comment by teriyaki-89** [comment]
> use --inspect-brk

you're magician

**Comment by mm41707** [comment]
> * find a hex editor, open `node.exe`, search for `portIsFree`, currently 2 results comes up, one of them has `timeout = 2000` nearby, replace it with a higher value, I suggest `9999`.

Hey! Great job.  I didn't even expect to find any plain text parameters inside win PE file. Great clean solution, without any 3rd party modules. That's what i was hoping to get. Cheers man!

**Comment by Ang3lExtreme** [comment]
> @mikemaccana Today, I am in a bit of debugger/patcher mood, so here is 4th, but surely unsupported method. After this, `node inspect` works without timeout error.
> 
> * find a hex editor, open `node.exe`, search for `portIsFree`, currently 2 results comes up, one of them has `timeout = 2000` nearby, replace it with a higher value, I suggest `9999`.
> 
> While doing that, you will notice that Node.js actually holds packages almost as is. If you look at the beginning of the file, you will see the "native" codes. Native codes are only read by machine logic (assembler). In contrast, packages are hold as their human readable form.
> 
> Actually I didn't expect this result. Anyone may apply this method on their free will, just should not expect support if something else gets broken.

you are a god!

**Comment by raj7096** [comment]
any Of This is not working

**Comment by r0otHex** [comment]
try this:

`node --inspect-brk app.js`

it works fine with me.

**Comment by rajithaeyee** [comment]
node --inspect-brk app.js solves the problem

**Comment by raj7096** [comment]
Thanks Now its Working😅

**Comment by dwkrueger2** [comment]
We need to move the 
node --inspect-brk app.js to the top of this feed instead of the bottom.

**Comment by yilmazdurmaz** [comment]
"--inspect" and "--inspect-brk" does things in slight different ways, and depending on what you need the second does the jobs.

we have a fix about the "timeout" for "--inspect" already committed&merged, so possibly the next node version change will include it. as for the fix, it is an increased value on the setTimeout function, 2000 to 9999, because the problem here is that Windows is somehow slow on taking the requested port, about 3-5 seconds, thus timeout was happening.

keep an eye on it and update your node installation when the fixed one comes out.

**Comment by jkrems** [comment]
Update: A change increasing the timeout has landed here and there's an open PR to pull it into node (https://github.com/nodejs/node/pull/33447).

**Comment by chumaachike** [comment]
> Error: Timeout (2000) waiting for 127.0.0.1:9229 to be free
> I have the same question

try node --inspect-brk filename.js

**Comment by HarPri2020** [comment]
Still I am facing issue with debugging. Didn't worked for me. Using node version v12.18.0. 
node --inspect-brk index.js
Debugger listening on ws://127.0.0.1:9229/93410f83-bdd1-488d-bcc9-595904b41697
For help, see: https://nodejs.org/en/docs/inspector

**Comment by vivekvking** [comment]
Go to your chrome browser and enter the url: chrome://inspect
then you will find something under remote target, then just click on the inspect option given over there and you are good to go.
If you donot find anything under remote targets then click upon configure option there and enter the ip that you got on the console that is 127.0.0.1:9229

**Comment by HarPri2020** [comment]
@vivekvking , i have updated chrome://inspect and added configuration - discover network targets as 127.0.0.1:9229. But the issue resists to   
Error: Timeout (2000) waiting for 127.0.0.1:9229 to be free
then realized, under https://nodejs.org/en/docs/guides/debugging-getting-started/  for latest version we need to install independently [npm install -g node-inspect] and  [node-inspect myscript.js] worked for me.

**Comment by sagartyagi121** [comment]
If you want to debug your code use   ` node --inspect yourJs.js` 
                                                    not   ` node inspect yourJS.js`

**Comment by juniorbuba** [comment]
Tried everything and none worked but
```node --inspect app```

**Comment by jochri3** [comment]
> @mikemaccana Today, I am in a bit of debugger/patcher mood, so here is 4th, but surely unsupported method. After this, `node inspect` works without timeout error.
> 
> * find a hex editor, open `node.exe`, search for `portIsFree`, currently 2 results comes up, one of them has `timeout = 2000` nearby, replace it with a higher value, I suggest `9999`.
> 
> While doing that, you will notice that Node.js actually holds packages almost as is. If you look at the beginning of the file, you will see the "native" codes. Native codes are only read by machine logic (assembler). In contrast, packages are hold as their human readable form.
> 
> Actually I didn't expect this result. Anyone may apply this method on their free will, just should not expect support if something else gets broken.

You're great❤️❤️❤️

**Comment by jbrandona119** [comment]
Typed up a whole gosh darn thing and after reading the docs I got it working finally! After installing globally with "npm install -g node-inspect" and restarting the terminal, I was able to use "**node-inspect fileName.js**" and it successfully starts and keeps the debugger running in that terminal. I suggest trying that if you're still struggling. I did try "node inspect" and "node --inspect" but they didn't work. What a strange issue!

**Comment by yilmazdurmaz** [comment]
Hey there all **future newcomers** to the topic (and some old still struggling).

**TL:DR** - we have many possible solutions. please try to read comments.

I wish comments were not this long and we had the habit of reading long things, impossible by our nature.

just know that, we know the problem and we have the solutions.

**PROBLEM**: newer version of windows is slow to assign debug port that takes about 3-5 seconds, but this debug utility had a predefined (working in the past) 2 seconds timeout. this is the only reason we have this issue.

**SOLUTION**: we already have the fix submitted to Node.JS codebase, and sometime in the future, depending on Node's own release schedule. until then there are many working solutions depending on your expectation, so please, please, please read these comments and try solutions.

thanks for patience

PS: the solution in above comment works now without any modification because we really have it fixed here. two post above works too because of the nature of the problem; we need a simple timeout increase.

PS: I really hope new readers will at least read the last post in the line, and keep this one as the last post.

**Comment by harikanani** [comment]
> To clarify, the command to start the CLI debugger on a custom port is `node inspect --port=9230 path/to/script.js`.
> 
> When using `node --inspect`, the error above shouldn't appear since that's the program being debugged, it's not starting the CLI debugger (`node-inspect`).

I'm having 


> i'm having this issue as well and only have one terminal up still has same error
> Node.js:12.2.0
> OS: Windows 10
> on VSC
> 
> ```
> $ node inspect app.js
> Timeout (2000) waiting for 127.0.0.1:9229 to be free
> ```

# Same Issue is there but this Command Works for Mine.
```node --inspect=127.0.0.1:9230 app_name.js```
# Thanks

**Comment by yilmazdurmaz** [comment]
Hi all

Latest Node v12.19.0 and v15.0.1 now has the fix in them (timeout set to 9999ms). Please update your installations when you are ready.

@jkrems , this #48 and #54 issues should be safe to be marked as solved.
